saleor/graphql/discount/mutations.py

```diff
@@ -270,7 +270,11 @@ class SaleUpdateMinimalVariantPriceMixin:
     def success_response(cls, instance):
         # Update the "minimal_variant_prices" of the associated, discounted
         # products (including collections and categories).
-        update_products_minimal_variant_prices_of_discount_task.delay(instance.pk)
+        products_exists = instance.products.all().exists()
+        categories_exists = instance.categories.all().exists()
+        collections_exists = instance.collections.all().exists()
```

Comment:
I wonder, `instance.products.all() or instance.categories.all() or ...` would resolve to `exists()` to make it lazy?